src/disco/components/Addon/index.js

```diff
@@ -121,31 +124,48 @@ export class AddonBase extends React.Component {
   }
 
   getThemeImage() {
-    const { getBrowserThemeData, i18n, name, previewURL } = this.props;
-    if (this.props.type === ADDON_TYPE_THEME) {
+    const { type } = this.props.addon;
+
+    if (ADDON_TYPE_THEMES.includes(type)) {
+      const { addon, getBrowserThemeData, i18n } = this.props;
```

Comment:
Would it make sense to get the addon above? (You use `this.props.addon`) to retrieve the `type`, but maybe we can get the `addon` first, and then its type.